File: docs/code-examples/segmentation_image_simple.py
"""Create and log a segmentation image."""

import numpy as np
import rerun as rr
import rerun.experimental as rr2

# Create a segmentation image
image = np.zeros((8, 12), dtype=np.uint8)
image[0:4, 0:6] = 1
image[4:8, 6:12] = 2

rr.init("rerun_example_segmentation_image", spawn=True)

# Assign a label and color to each class
rr2.log("/", rr2.AnnotationContext([(1, "red", (255, 0, 0)), (2, "green", (0, 255, 0))]))

# TODO(#2792): SegmentationImage archetype
rr.log_segmentation_image("image", np.array(image))

File: docs/code-examples/segmentation_image_simple.rs
//! Create and log a segmentation image.

use ndarray::{s, Array, ShapeBuilder};
use rerun::datatypes::Color;

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let (rec, storage) =
        rerun::RecordingStreamBuilder::new("rerun_example_segmentation_image").memory()?;

    // create a segmentation image
    let mut image = Array::<u8, _>::zeros((8, 12).f());
    image.slice_mut(s![0..4, 0..6]).fill(1);
    image.slice_mut(s![4..8, 6..12]).fill(2);

    // create an annotation context to describe the classes
    let annotation = rerun::AnnotationContext::new([
        (1, "red", Color::from(0xFF0000FF)),
        (2, "green", Color::from(0x00FF00FF)),
    ]);

    // log the annotation and the image
    rec.log("/", &annotation)?;

    rec.log("image", &rerun::SegmentationImage::try_from(image)?)?;

    rerun::native_viewer::show(storage.take())?;
    Ok(())
}
